web-console/src/components/timed-button/timed-button.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Button, ButtonGroup, IButtonProps, Popover, Radio, RadioGroup } from '@blueprintjs/core';
+import { IconNames } from '@blueprintjs/icons';
+import React from 'react';
+
+import { localStorageGet, LocalStorageKeys, localStorageSet } from '../../utils';
+
+import './timed-button.scss';
+import Timeout = NodeJS.Timeout;
+
+export interface Interval {
+  label: string;
+  value: number;
+}
+
+export interface TimedButtonProps extends IButtonProps {
+  intervals: Interval[];
+  onRefresh: (auto: boolean) => void;
+  localstoragekey?: LocalStorageKeys;
+  label: string;
+}
+
+export interface TimedButtonState {
+  interval: number;
+}
+
+export class TimedButton extends React.PureComponent<TimedButtonProps, TimedButtonState> {
+  constructor(props: TimedButtonProps, context: any) {
+    super(props, context);
+    this.state = {
+      interval: 30000
+    };
+  }
+
+  private timer: Timeout;
+
+  componentDidMount(): void {
+    let interval = 30000;
+    if (this.props.localstoragekey) {
+      if (localStorageGet(this.props.localstoragekey) !== null) {
+        interval = Number(localStorageGet(this.props.localstoragekey));
+      }
+      this.setState({interval: interval});
+      if (interval) {
+        this.timer = setTimeout(() => {this.continousRefresh(interval); }, interval);
+      }
+    }
+  }
+
+
+  componentWillUnmount(): void {
+    clearTimeout(this.timer);
+  }
+
+  continousRefresh = (selectedInterval: number) => {
+    if (selectedInterval) {
+      this.timer = setTimeout(() => {
+        this.props.onRefresh(true);
+        this.continousRefresh(selectedInterval);
+      }, selectedInterval);
+    }
+  }
+
+  handleSelection( selectedInterval: number) {
+    clearTimeout(this.timer);
+    this.setState({interval: selectedInterval});
+    if (this.props.localstoragekey) {
+      localStorageSet(this.props.localstoragekey, String(selectedInterval));
+    }
+    this.continousRefresh(selectedInterval);
+  }
+
+  render() {
+    const { label, intervals, localstoragekey, onRefresh, type, ...other } = this.props;
+    const { interval } = this.state;
+
+    return <ButtonGroup>
```

Comment:
for completeness you might need to pass some props (like `minimal`, `large`, `intent`, etc) to the ButtonGroup. Have a look at the button group interface and see what props it shares with Button and make sure those all work as desired.

Reply:
<img width="537" alt="Screen Shot 2019-06-19 at 8 27 45 AM" src="https://user-images.githubusercontent.com/37322608/59779310-dfc71400-926c-11e9-8e23-068d010e6ffd.png">
<img width="491" alt="Screen Shot 2019-06-19 at 8 28 52 AM" src="https://user-images.githubusercontent.com/37322608/59779311-dfc71400-926c-11e9-8039-ffba12db7cff.png">

It should be okay now. I changed it so now all the properties but icon and text are given to both buttons now. That means anything you apply to one button that is originally a property of IButton is given to both.